src/lib/components/healthselector/Healthselector.component.js

```diff
@@ -0,0 +1,322 @@
+					width: 190px;
+					ul {
+						width: 190px;
+						li {
+							padding: ${defaultTheme.padding.larger};
+						}
+					}
+				`;
+			default:
+				return css`
+					${CSSCircle} {
+						width: 11px;
+						height: 11px;
+					}
+
+					padding: ${defaultTheme.padding.base};
+					width: 130px;
+					ul {
+						width: 130px;
+						li {
+							padding: ${defaultTheme.padding.base};
+						}
+					}
+				`;
+		}
+	}}
+`;
+
+const LeftRowWrapper = styled.div`
+	display: flex;
+	justify-content: flex-end;
+	width: 40%;
+`;
+const RightRowWrapper = styled.div`
+	width: 60%;
+	text-align: left;
+	padding-left: 10%;
+`;
+
+const HealthSelectorMenu = styled.ul`
+	box-sizing: border-box;
+	position: absolute;
+	margin: 0;
+	padding: 0;
+	z-index: ${defaultTheme.zIndex.dropdown};
+
+	${(props) => {
+		if (
+			props.size &&
+			props.triggerSize &&
+			props.triggerSize.x + props.size.width > window.innerWidth
+		) {
+			return css`
+				right: 0;
+				top: 100%;
+			`;
+		} else if (
+			props.size &&
+			props.triggerSize &&
+			props.triggerSize.y + props.size.height > window.innerHeight
+		) {
+			return css`
+				left: 0;
+				bottom: ${props.triggerSize.height + 'px'};
+			`;
+		} else {
+			return css`
+				left: 0;
+				top: 100%;
+			`;
+		}
+	}};
+`;
+
+const HealthSelectorMenuItem = styled.li`
+	display: flex;
+	align-items: center;
+	white-space: nowrap;
+	cursor: pointer;
+	justify-content: space-around;
+
+	${css`
+		background-color: ${getThemePropSelector('primary')};
+		color: ${getThemePropSelector('textPrimary')};
+		&:hover {
+			background-color: ${getThemePropSelector('primaryDark2')};
+		}
+		&:active {
+			background-color: ${getThemePropSelector('primaryDark2')};
+		}
+	`};
+`;
+
+function Healthselector(props: Props) {
+	const { variant, size, items } = props;
+	const [open, setOpen] = useState(false);
+	const [menuSize, setMenuSize] = useState();
+	const [triggerSize, setTriggerSize] = useState();
+	const selected = items.find((item) => item.selected);
+	const selectedIndex =
+		items.indexOf(selected) >= 0 ? items.indexOf(selected) : 0;
+
+	const refMenuCallback = useCallback((node) => {
+		if (node !== null) {
+			setMenuSize(node.getBoundingClientRect());
+		}
+	}, []);
+
+	const refTriggerCallback = useCallback((node) => {
+		if (node !== null) {
+			setTriggerSize(node.getBoundingClientRect());
+		}
+	}, []);
+
+	const icons = [
+		<ThreeCirclesWrapper>
+			<CSSCircle variant="healthy" />
+			<CSSCircle variant="warning" />
+			<CSSCircle variant="danger" />
+		</ThreeCirclesWrapper>,
+		<CircleWrapper>
+			<CSSCircle variant="healthy" />
+		</CircleWrapper>,
+		<CircleWrapper>
+			<CSSCircle variant="warning" />
+		</CircleWrapper>,
+		<CircleWrapper>
+			<CSSCircle variant="danger" />
+		</CircleWrapper>,
+	];
+
+	return (
+		<HealthselectorContainer className="sc-healthselector">
+			<TriggerStyled
+				variant={variant || 'base'}
+				size={size || 'base'}
+				className="trigger"
```

Comment:
the className would be better to rename to `sc-healthselector-trigger`, so it's kind of unique.